components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodProjectManager.kt

```diff
@@ -29,20 +33,37 @@ class GitpodProjectManager(
         configureSdks()
     }
 
-    /**
-     * It is a workaround for https://youtrack.jetbrains.com/issue/GTW-88
-     */
     private fun configureSdks() {
         if (application.isHeadlessEnvironment) {
             return
         }
         val pendingSdk = CompletableFuture<Sdk>()
         application.invokeLaterOnWriteThread {
             application.runWriteAction {
+                val jdkTable = ProjectJdkTable.getInstance()
+
                 try {
-                    ProjectJdkTable.getInstance().preconfigure()
-                    pendingSdk.complete(ProjectJdkTable.getInstance().allJdks.firstOrNull())
+                    var sdk: Sdk? = null
+                    val javaHome = JavaHomeFinder.getFinder().findExistingJdks().firstOrNull()
+                    val sdkType = SdkType.findByName(jdkTable.defaultSdkType.name)
+
+                    if (javaHome !== null && sdkType != null && JdkUtil.checkForJdk(javaHome)) {
+                        val javaHomeVirtualFile = VirtualFileManager.getInstance().findFileByNioPath(Path.of(javaHome))
+
+                        if (javaHomeVirtualFile != null) {
+                            sdk = SdkConfigurationUtil.setupSdk(jdkTable.allJdks, javaHomeVirtualFile, sdkType, false, null, null)
+                        }
+                    }
+
+                    if (sdk != null && jdkTable.findJdk(sdk.name) == null) {
+                        jdkTable.addJdk(sdk)
+                        pendingSdk.complete(sdk)
+                    } else {
+                        application.invokeAndWait { jdkTable.preconfigure() }
```

Comment:
I added `application.invokeAndWait` cause I thought it would solve [this issue](https://github.com/gitpod-io/gitpod/pull/12163#issuecomment-1218251482).
As I confirmed the issue was unrelated to it, I'll remove it in my next cleanup-commit.